js/assessments/subheadingDistributionTooLongAssessment.js

```diff
@@ -0,0 +1,102 @@
+var AssessmentResult = require( "../values/AssessmentResult.js" );
+var filter = require( "lodash/filter" );
+
+/**
+ *  Returns true or false, based on the length of the subheading text and the recommended value.
+ * @param {number} recommendedValue The recommended value of subheading text length
+ * @param {number} subheadingsTextLength The length of the text after the subheading
+ * @returns {boolean} True if the length is greater than the recommendedValue,
+ * false if it is smaller.
+ */
+var isTextTooLong = function( recommendedValue, subheadingsTextLength ) {
+	return subheadingsTextLength > recommendedValue;
+};
+
+/**
+ * Counts the number of subheading texts that are too long.
+ * @param {Array} subheadingTextsLength Array with subheading text lengths.
+ * @param {number} recommendedValue The recommended maximum value of the subheading text length
+ * @returns {number} The number of subheading texts that are too long.
+ */
+var getTooLongSubheadingTexts = function( subheadingTextsLength, recommendedValue ) {
+	var tooLongTexts = filter( subheadingTextsLength, isTextTooLong.bind( null, recommendedValue ) );
+	return tooLongTexts.length;
+};
+
+/**
+ * Calculates the score based on the subheading texts length.
+ * @param {Array} subheadingTextsLength Array with subheading text lengths.
+ * @param {number} tooLongTexts The number of subheading texts that are too long.
+ * @param {number} recommendedValue The recommended maximum value of the subheading text length
+ * @param {object} i18n The object used for translations.
+ * @returns {object} the resultobject containing a score and text if subheadings are present
+ */
+var subheadingsTextLength = function( subheadingTextsLength, tooLongTexts, recommendedValue, i18n ) {
+
+	// Return empty result if there are no subheadings
+	if ( subheadingTextsLength.length === 0 ) {
+		return {};
+	}
+
+	// 6 is the number of scorepoints between 3, minscore and 9, maxscore. For scoring we use 100 steps, each step is 0.06.
+	// Up to 267  is for scoring a 9, higher numbers give lower scores.
+	var score = 9 - Math.max( Math.min( ( 0.06 ) * ( subheadingTextsLength[ 0 ] - 267 ), 6 ), 0 );
```

Comment:
Maybe we should create a function for this, because we use this kind of algorithm multiple times but with different values. I think it should become something like this

```
function calculateScoreOrGiveItAnotherName( text, threshold  ) {
    return 9 - Math.max( Math.min( ( 0.06 ) * ( text - treshhold ), 6 ), 0 );
```

and call it by:
`calculateScoreOrGiveItAnotherName( subheadingTextsLength[ 0 ], 267  );`